src/librustc_driver/driver.rs

```diff
@@ -751,14 +751,18 @@ pub fn phase_2_configure_and_expand<F>(sess: &Session,
          || ast_validation::check_crate(sess, &krate));
 
     time(time_passes, "name resolution", || -> CompileResult {
-        // Since import resolution will eventually happen in expansion,
-        // don't perform `after_expand` until after import resolution.
+        // Since import resolution happs in expansion, don't perform
+        // `after_expand` until after import resolution.
```

Comment:
I would just remove this comment -- since expansion and import resolution are interleaved, there's no way to perform `after_expand` before import resolution, so we don't need to warn people about this.

Also, we can move `after_expand(&krate)?` outside of  `time(time_passes, "name resolution", ...)` now, I think it would be best to put it just before expansion debugging diagnostics (`"Post-expansion node count"`, etc.).